### ISSUE TITLE
Notifications not being translated

### ISSUE THREAD

**Issue by seadowg** [getodk/collect#3988]
~~Blocked by #3986.~~

#### Software and hardware versions 

Collect v1.27.3 (older versions as well)

#### Problem description

It seems notifications for auto send and auto update are not being translated.

#### Steps to reproduce the problem

1. Set Collect's language to something other than English
1. Enabled auto send
1. Fill out and finalize a form

You'll see the notification is in English.

#### Expected behavior

Notifications use the same language as the rest of the app.

#### Other information 

This will be problem wherever we use the `Application` to grab resource strings rather than an `Activity`. This is because the way we change the language only does so for `Activity` contexts, but we're unable to change the resource locale at an `Application` level. We're seeing this in Notifications as they are mostly created from background jobs that would use the application context as they don't have access to an `Activity`. 

#3986 introduces code that lets us create resource objects with the correct locale from the application context (in `NotificationManagerNotifier`). Once that is merged we can use that for all the notifications.

**Comment by grzesiek2010** [comment]
It's a common problem in our app and there are other reported issues like: https://github.com/getodk/collect/issues/3029
In some cases we just can't use activity context.
How about creating a helper class for dealing with strings?
```java
public class TranslationHandler {
    public static String getString(int stringId) {
        return getLocalizedResources(Collect.getInstance()).getString(stringId);
    }

    // The application context will give us the system's locale
    private static Resources getLocalizedResources(Context context) {
        Configuration conf = context.getResources().getConfiguration();
        conf = new Configuration(conf);
        conf.setLocale(new Locale(LocaleHelper.getLocaleCode(context)));
        Context localizedContext = context.createConfigurationContext(conf);
        return localizedContext.getResources();
    }
}
```

and calling:
```java
TranslationHandler.getString(R.string.sting_name);
```
across the app?

**Comment by seadowg** [comment]
Yeah I like that! I think it'd be nicer for it to take in `Context` as well as a `int` string id, so we're not relying on `Collect.getInstance()`.

**Comment by grzesiek2010** [comment]
> I think it'd be nicer for it to take in Context as well as a int string id, so we're not relying on Collect.getInstance().

why do you find it important? Then every call will look like:
```java
TranslationHandler.getString(getActivity(), R.string.sting_name);
TranslationHandler.getString(getContext(), R.string.sting_name);
TranslationHandler.getString(Collect.getInstance(), R.string.sting_name);
```
depending where it is called from. If the result is the same for all cases and we can relay on application context only why not to avoid passing context at all?

**Comment by seadowg** [comment]
> why do you find it important?

Just for modularity. If it relies on `Collect.getInstance()` it creates a hard dependency on the `Collect`. This isn't a huge problem for this method but anything that uses it now depends on `Collect`. Imagine if I want to separate out some component (like the notifier or a view) into a submodule: I'd have to rework it, so I don't end up with a circular dependency between the new submodule and `collect_app`.

In short, it just means we end up with a higher degree of coupling than we need. Not the end of the world at but nice to avoid in my opinion.

**Comment by grzesiek2010** [comment]
I understand the concept of course but I'm also not a fan of modulating everything. For example this particular methods seems to me like something that could depend on the application context because it will always work and it will always be available.
another question is do we want to refactor all cases where translations are used using the new class or just those where activity context is not available and we use the application context either way? If we go for the second scenario we can have one method like the one I created because in all cases it will be the application context and we can create a separate one which takes context into account later (or even now but not use) to use in all the other cases? Does that make sense?

**Comment by seadowg** [comment]
> For example this particular methods seems to me like something that could depend on the application context because it will always work and it will always be available.

It's not relying on the `Application`'s context that bothers me here: it's relying on `Collect.getInstance()` and the resulting dependency on `Collect`. It just feels like unnecessary coupling to have the method have to know about our implementation of `Application`. 

I could totally be missing something here but as far as I understand it the implementation in `getLocalizedResources()` doesn't need the context to be a `Collect` or even an `Application` instance and anywhere trying to access a string to display would most likely have access to `Context` object (be it an `Activity` or an `Application`), so I don't really see the advantage to the hard dependency on `Collect` versus passing in the context at the call site.

That said, it's not the end of the world to have to extract the context to a parameter later if we need to, so I don't want it to be a blocker.

> another question is do we want to refactor all cases where translations are used using the new class or just those where activity context is not available and we use the application context either way?

Yeah refactoring everything in one go would probably be nasty. Prioritizing the places we only have the application context definitely feels like the right move 👍. We could then add a note to our style guide and just try and convert other places as we go?
